packages/next/src/server/lib/start-server.ts

```diff
@@ -257,6 +260,11 @@ export async function startServer(
         const startServerInfo = await getStartServerInfo(dir, isDev)
         envInfo = startServerInfo.envInfo
         expFeatureInfo = startServerInfo.expFeatureInfo
+
+        // opt out if typedEnv is set to false
+        if (startServerInfo.nextConfig.typedEnv !== false) {
```

Comment:
this will be duplicated, it will executed on setup-dev-bundler in dev. We don't need it here

Reply:
`setup-dev-bundler` runs on `envChange` for HMR, should I add some logic there to run initially?

x-ref: [here](https://github.com/vercel/next.js/pull/67157/files#diff-bd156981b29b52a48423333c2c84bcf012bd98d4e0a5ae7de0f78fc4297aba46R547-R550)

Reply:
the one in setup-dev is going to run anyway for initial start, we don't need to call it here. 
If your purpose is to do type check in build mode, then might need to parse the env file and write before build TS check started.